app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by email: params[:email]
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
+      user.activate
+      log_in user
+      flash[:success] = t ".account_activated"
+      redirect_to user
+    else
+      flash[:danger] = t ".invalid_activation_link"
+      redirect_to root_url
```

Comment:
you should use `path` instead of `url`
`You don't need to use *_url for controllers, path will work just as well. You should only use _url when displaying the route to outside sources such as link send mail`